tests/nested/manual/grade-signed-cloud-init-testkeys/task.yaml

```diff
@@ -0,0 +1,167 @@
+  NESTED_CUSTOM_AUTO_IMPORT_ASSERTION: $TESTSLIB/assertions/developer1-auto-import.assert
+
+  NESTED_CUSTOM_MODEL: $TESTSLIB/assertions/developer1-20-signed.model
+  NESTED_IMAGE_ID: cloud-init-signed-testkeys
+
+  # for the fake store
+  NESTED_FAKESTORE_BLOB_DIR: $(pwd)/fake-store-blobdir
+  NESTED_UBUNTU_IMAGE_SNAPPY_FORCE_SAS_URL: http://localhost:11028
+
+  # unset this otherwise ubuntu-image complains about overriding the channel for
+  # a model with grade higher than dangerous when building the image
+  NESTED_CORE_CHANNEL: ""
+
+prepare: |
+  if [ "$TRUST_TEST_KEYS" = "false" ]; then
+      echo "This test needs test keys to be trusted"
+      exit
+  fi
+
+  #shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+
+  #shellcheck source=tests/lib/store.sh
+  . "$TESTSLIB"/store.sh
+
+  # setup the fakestore, but don't use it for our snapd here on the host VM, so
+  # tear down the staging_store immediately afterwards so that only the SAS is 
+  # running and our snapd is not pointed at it, ubuntu-image is the only thing 
+  # that actually needs to use the fakestore, and we will manually point it at
+  # the fakestore below using NESTED_UBUNTU_IMAGE_SNAPPY_FORCE_SAS_URL
+  setup_fake_store "$NESTED_FAKESTORE_BLOB_DIR"
+  teardown_staging_store
+
+  echo Expose the needed assertions through the fakestore
+  cp "$TESTSLIB"/assertions/developer1.account "$NESTED_FAKESTORE_BLOB_DIR/asserts"
+  cp "$TESTSLIB"/assertions/developer1.account-key "$NESTED_FAKESTORE_BLOB_DIR/asserts"
+
+  # modify and repack gadget snap to add a defaults section and use our own
+  # prepare-device hook to use the fakedevicesvc
+  mkdir "$(nested_get_extra_snaps_path)"
+
+  # Get the snakeoil key and cert for signing gadget assets (shim)
+  KEY_NAME=$(nested_get_snakeoil_key)
+  SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+  SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+
+  snap download --basename=pc --channel="20/edge" pc
+  unsquashfs -d pc-gadget pc.snap
+
+  # delay all refreshes for a week from now, as otherwise refreshes for our 
+  # snaps (which are asserted by the testrootorg authority-id) may happen, which
+  # will break things because the signing keys won't match, etc. and 
+  # specifically snap-bootstrap in the kernel snap from the store won't trust
+  # the seed keys to unlock the encrypted data partition in the initramfs
+  sed defaults.yaml -e "s/HOLD-TIME/$(date --date='next week' +%Y-%m-%dT%H:%M:%S%:z)/" >> \
+    pc-gadget/meta/gadget.yaml
+
+  # TODO: enable this bit when things are ready to use a testkeys signed model
+  # assertion
+  # copy the prepare-device hook to use our fakedevicesvc
+  # mkdir -p pc-gadget/meta/hooks/
+  # cp prepare-device pc-gadget/meta/hooks/
+
+  nested_secboot_sign_gadget pc-gadget "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  snap pack pc-gadget/ extra-snaps/
+  rm -rf pc-gadget/
+
+  rm -f "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+
+  # TODO: enable when ready
```

Comment:
It is unclear when this will be ready. If you have a moment please revise this paragraph.

Reply:
sure I hope to do it $SOON, is that clear enough ? :smile:

Reply:
elaborated in the comment